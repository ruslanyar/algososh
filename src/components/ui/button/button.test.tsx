import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Button } from './button';

describe('Компонент Button', () => {
  it('Компонент есть в документе', () => {
    render(<Button />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
  });

  it('Snapshot: кнопка с текстом', () => {
    const tree = renderer.create(<Button text='Click me' />);
    expect(tree).toMatchSnapshot();
  });

  it('Snapshot: кнопка без текста', () => {
    const tree = renderer.create(<Button />);
    expect(tree).toMatchSnapshot();
  });

  it('Snapshot: заблокированная кнопка', () => {
    const tree = renderer.create(<Button disabled />);
    expect(tree).toMatchSnapshot();
  });

  it('Snapshot: кнопка с индикацией загрузки', () => {
    const tree = renderer.create(<Button isLoader />);
    expect(tree).toMatchSnapshot();
  });

  it('Нажатие на кнопку вызывает колбек', () => {
    const onClickCb = jest.fn();
    render(<Button onClick={onClickCb}/>);
    const btn = screen.getByRole('button');
    fireEvent.click(btn);
    expect(onClickCb).toHaveBeenCalledTimes(1);
  });
});