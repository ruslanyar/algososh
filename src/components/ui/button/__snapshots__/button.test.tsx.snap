// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Button Snapshot: заблокированная кнопка 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Компонент Button Snapshot: кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Компонент Button Snapshot: кнопка с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Компонент Button Snapshot: кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Click me
  </p>
</button>
`;
